tests/shared/src/test/scala/coursier/test/ParseTests.scala

```diff
@@ -86,6 +86,32 @@ object ParseTests extends TestSuite {
       }
     }
 
+    "single attr with url" - {
+      Parse.moduleVersionConfig("org.apache.avro:avro:1.7.4:runtime,url=file%3A%2F%2Fsome%2Fencoded%2Furl", ModuleRequirements(), transitive = true, "2.11.11") match {
+        case Left(err) => assert(false)
+        case Right(dep) =>
+          assert(dep.module.organization == "org.apache.avro")
+          assert(dep.module.name == "avro")
+          assert(dep.version == "1.7.4")
+          assert(dep.configuration == "runtime")
+          assert(dep.attributes == Attributes("", "", "file://some/encoded/url"))
```

Comment:
How about making `"file://some/encoded/url"` a variable, and then make the moduleVersionConfig to be `"org.apache.avro:avro:1.7.4:runtime,url=" +  encoded(...)`?

Same for below